### ISSUE TITLE
racket's reimplementation of paredit-open-bracket breaks prefix arguments

### ISSUE THREAD

**Issue by jaor** [greghendershott/racket-mode#378]
When not intercepted by racket, paredit-open-bracket ([) accepts
prefix arguments say `M-3 [`, causing the `[]` to span, in this
example, 3 sexps.  Racket's model rewriting of that function **for
other modes** (well, at least in clojure, i haven't tried anything
else) doesn't seem to be forwarding the prefix argument and breaks
that behaviour: the prefix argument is simply ignored.

Thanks!

<details>
```
((alist-get 'racket-mode package-alist)
 #s(package-desc racket-mode
                 (20190607 109)
                 "Major mode for Racket language."
                 ((emacs
                   (24 3))
                  (faceup
                   (0 0 2)))
                 nil nil "/home/jao/.emacs.d/elpa.26/racket-mode-20190607.109"
                 ((:url . "https://www.racket-mode.com/")
                  (:maintainer "Greg Hendershott")
                  (:authors
                   ("Greg Hendershott")))
                 nil))
((emacs-version "26.2.50")
 (emacs-uptime "3 days, 4 hours, 31 minutes, 42 seconds")
 (system-type gnu/linux)
 (major-mode scheme-mode)
 (racket--el-source-dir "/home/jao/.emacs.d/elpa.26/racket-mode-20190607.109/")
 (racket--rkt-source-dir "/home/jao/.emacs.d/elpa.26/racket-mode-20190607.109/racket/")
 (racket-program "racket")
 (racket-command-port 55555)
 (racket-command-startup nil)
 (racket-command-timeout 10)
 (racket-memory-limit 2048)
 (racket-error-context medium)
 (racket-retry-as-skeleton t)
 (racket-error-context medium)
 (racket-history-filter-regexp "\\`\\s *\\S ?\\S ?\\s *\\'")
 (racket-images-inline t)
 (racket-images-keep-last 100)
 (racket-use-repl-submit-predicate nil)
 (racket-images-system-viewer "display")
 (racket-pretty-print t)
 (racket-indent-curly-as-sequence t)
 (racket-indent-sequence-depth 0)
 (racket-pretty-lambda nil)
 (racket-smart-open-bracket-enable nil)
 (racket-module-forms "\\s(\\(?:module[*+]?\\|library\\)")
 (racket-logger-config
  ((cm-accomplice . warning)
   (GC . info)
   (module-prefetch . warning)
   (optimizer . info)
   (racket/contract . error)
   (sequence-specialization . info)
   (* . fatal))))
(enabled-minor-modes
 (async-bytecomp-package-mode)
 (auto-composition-mode)
 (auto-compression-mode)
 (auto-encryption-mode)
 (auto-fill-mode)
 (auto-save-mode)
 (circe-lagmon-mode)
 (column-number-mode)
 (company-mode)
 (counsel-mode)
 (diff-auto-refine-mode)
 (diff-hl-mode)
 (display-time-mode)
 (eldoc-mode)
 (electric-indent-mode)
 (fci-mode)
 (file-name-shadow-mode)
 (font-lock-mode)
 (geiser-autodoc-mode)
 (geiser-mode)
 (global-auto-revert-mode)
 (global-company-mode)
 (global-diff-hl-mode)
 (global-edit-server-edit-mode)
 (global-eldoc-mode)
 (global-font-lock-mode)
 (global-git-commit-mode)
 (global-magit-file-mode)
 (intero-global-mode)
 (ivy-mode)
 (ivy-rich-mode)
 (jao-treemacs-follow-mode)
 (line-number-mode)
 (magit-auto-revert-mode)
 (mouse-wheel-mode)
 (override-global-mode)
 (paredit-mode)
 (pdf-occur-global-minor-mode)
 (persistent-scratch-autosave-mode)
 (recentf-mode)
 (save-place-mode)
 (savehist-mode)
 (semantic-minor-modes-format)
 (shell-dirtrack-mode)
 (show-paren-mode)
 (tooltip-mode)
 (tracking-mode)
 (treemacs-filewatch-mode)
 (treemacs-fringe-indicator-mode)
 (treemacs-git-mode)
 (winner-mode))
(disabled-minor-modes
 (abbrev-mode)
 (ace-window-display-mode)
 (ace-window-mode)
 (archive-subfile-mode)
 (auto-fill-function)
 (auto-image-file-mode)
 (auto-insert-mode)
 (auto-revert-mode)
 (auto-revert-tail-mode)
 (auto-save-visited-mode)
 (avy-linum-mode)
 (blink-cursor-mode)
 (buffer-read-only)
 (bug-reference-mode)
 (bug-reference-prog-mode)
 (cider--debug-mode)
 (cider-auto-test-mode)
 (cider-enlighten-mode)
 (cider-macroexpansion-mode)
 (cider-mode)
 (cider-popup-buffer-mode)
 (cl-old-struct-compat-mode)
 (company-search-mode)
 (compilation-in-progress)
 (compilation-minor-mode)
 (compilation-shell-minor-mode)
 (completion-in-region-mode)
 (cursor-intangible-mode)
 (cursor-sensor-mode)
 (defining-kbd-macro)
 (delete-selection-mode)
 (diff-hl-dir-mode)
 (diff-minor-mode)
 (dired-hide-details-mode)
 (dired-isearch-filenames-mode)
 (dired-omit-mode)
 (diredp-breadcrumbs-in-header-line-mode)
 (doc-view-minor-mode)
 (ebib-entry-minor-mode)
 (ebib-multiline-mode)
 (edebug-mode)
 (edit-server-edit-mode)
 (electric-layout-mode)
 (electric-pair-mode)
 (electric-quote-mode)
 (emms-volume-minor-mode)
 (emojify-debug-mode)
 (emojify-mode)
 (emojify-mode-line-mode)
 (flycheck-mode)
 (flyspell-mode)
 (geiser-smart-tab-mode)
 (gist-mode)
 (git-commit-mode)
 (git-ps1-mode)
 (git-timemachine-mode)
 (global-dired-hide-details-mode)
 (global-emojify-mode)
 (global-emojify-mode-line-mode)
 (global-flycheck-mode)
 (global-hl-line-mode)
 (global-prettify-symbols-mode)
 (global-semantic-highlight-edits-mode)
 (global-semantic-highlight-func-mode)
 (global-semantic-show-parser-state-mode)
 (global-semantic-show-unmatched-syntax-mode)
 (global-semantic-stickyfunc-mode)
 (global-undo-tree-mode)
 (global-visual-line-mode)
 (gnus-agent-group-mode)
 (gnus-agent-summary-mode)
 (gnus-dead-summary-mode)
 (gnus-draft-mode)
 (gnus-mailing-list-mode)
 (gnus-message-citation-mode)
 (gnus-topic-mode)
 (gnus-undo-mode)
 (gud-tooltip-mode)
 (hl-line-mode)
 (horizontal-scroll-bar-mode)
 (hs-minor-mode)
 (html-autoview-mode)
 (ibuffer-auto-mode)
 (ido-cr+-debug-mode)
 (ido-everywhere)
 (ido-ubiquitous-mode)
 (image-dired-minor-mode)
 (image-minor-mode)
 (intero-highlight-uses-mode)
 (intero-mode)
 (isearch-mode)
 (ispell-minor-mode)
 (jit-lock-debug-mode)
 (macrostep-mode)
 (magit-blame-mode)
 (magit-blame-read-only-mode)
 (magit-blob-mode)
 (magit-file-mode)
 (magit-wip-after-apply-mode)
 (magit-wip-after-save-local-mode)
 (magit-wip-after-save-mode)
 (magit-wip-before-change-mode)
 (magit-wip-initial-backup-mode)
 (magit-wip-mode)
 (mail-abbrevs-mode)
 (markdown-live-preview-mode)
 (menu-bar-mode)
 (merlin-mode)
 (minibuffer-electric-default-mode)
 (mml-mode)
 (ncm-mode)
 (next-error-follow-minor-mode)
 (org-capture-mode)
 (org-cdlatex-mode)
 (org-src-mode)
 (org-table-follow-field-mode)
 (orgstruct-mode)
 (orgtbl-mode)
 (outline-minor-mode)
 (overwrite-mode)
 (paragraph-indent-minor-mode)
 (pdf-annot-edit-contents-minor-mode)
 (pdf-annot-list-follow-minor-mode)
 (pdf-annot-minor-mode)
 (pdf-cache-prefetch-minor-mode)
 (pdf-history-minor-mode)
 (pdf-isearch-active-mode)
 (pdf-isearch-batch-mode)
 (pdf-isearch-minor-mode)
 (pdf-links-minor-mode)
 (pdf-misc-context-menu-minor-mode)
 (pdf-misc-menu-bar-minor-mode)
 (pdf-misc-minor-mode)
 (pdf-misc-size-indication-minor-mode)
 (pdf-occur-dired-minor-mode)
 (pdf-occur-ibuffer-minor-mode)
 (pdf-outline-follow-mode)
 (pdf-outline-minor-mode)
 (pdf-sync-backward-debug-minor-mode)
 (pdf-sync-minor-mode)
 (pdf-view-auto-slice-minor-mode)
 (pdf-view-dark-minor-mode)
 (pdf-view-midnight-minor-mode)
 (pdf-view-printer-minor-mode)
 (pdf-view-restore-mode)
 (prettify-symbols-mode)
 (racket-check-syntax-mode)
 (rectangle-mark-mode)
 (semantic-highlight-edits-mode)
 (semantic-highlight-func-mode)
 (semantic-mode)
 (semantic-show-parser-state-mode)
 (semantic-show-unmatched-syntax-mode)
 (semantic-stickyfunc-mode)
 (server-mode)
 (sgml-electric-tag-pair-mode)
 (sh-electric-here-document-mode)
 (shell-command-with-editor-mode)
 (size-indication-mode)
 (slime-autodoc-mode)
 (slime-edit-value-mode)
 (slime-editing-mode)
 (slime-fuzzy-target-buffer-completions-mode)
 (slime-macroexpansion-minor-mode)
 (slime-mode)
 (slime-popup-buffer-mode)
 (slime-repl-map-mode)
 (slime-repl-read-mode)
 (slime-trace-dialog-autofollow-mode)
 (slime-trace-dialog-hide-details-mode)
 (slime-trace-dialog-minor-mode)
 (smerge-mode)
 (spaceline-helm-mode)
 (spaceline-info-mode)
 (table-fixed-width-mode)
 (table-mode-indicator)
 (tablist-edit-column-minor-mode)
 (tablist-minor-mode)
 (tar-subfile-mode)
 (temp-buffer-resize-mode)
 (thai-word-mode)
 (tool-bar-mode)
 (transient-mark-mode)
 (transient-resume-mode)
 (treemacs-follow-mode)
 (treemacs-tag-follow-mode)
 (undo-tree-mode)
 (undo-tree-visualizer-selection-mode)
 (unify-8859-on-decoding-mode)
 (unify-8859-on-encoding-mode)
 (url-handler-mode)
 (use-hard-newlines)
 (utop-minor-mode)
 (vc-parent-buffer)
 (view-mode)
 (visible-mode)
 (visual-line-mode)
 (w3m-bookmark-mode)
 (w3m-fb-mode)
 (w3m-form-input-textarea-mode)
 (w3m-lnum-mode)
 (w3m-minor-mode)
 (which-function-mode)
 (window-divider-mode)
 (with-editor-mode)
 (xref-etags-mode))
```
</details>

**Comment by greghendershott** [comment]
Thanks for the report!

I can reproduce: <kbd>M-3 [</kbd> doesn't work in clojure-mode or emacs-lisp-mode.

As a work-around, using `paredit-wrap-square` works: <kbd>M-3 M-[</kbd>. But of course you shouldn't need a work-around.

Poking at this briefly, I was able to get a quick fix by passing along the prefix argument.

However, more deeply, I'm trying to remember why I thought, years ago, I had to modify `paredit-mode-map`. Instead could Racket Mode's smart-bracket feature be a minor mode, with its own key map --- and let Emacs work as-designed? Maybe I was concerned that if two minor modes had bindings for the same key, it wouldn't work reliably. I'll have a think.

If I can fix it "more deeply", soon, I'll do that. Otherwise I'll tweak the current approach by passing along the prefix argument.
